src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -1180,6 +1180,13 @@ public RequestLineStatus TakeStartLine(SocketInput input)
 
                 return RequestLineStatus.Done;
             }
+            catch (DecodingException)
+            {
+                RejectRequest(RequestRejectionReason.NonAsciiOrNullCharactersInRequestLine,
+                    Log.IsEnabled(LogLevel.Information) ? start.GetAsciiStringEscaped(end, MaxInvalidRequestLineChars) : string.Empty);
+                // won't run but compiler complains
```

Comment:
Would it work if we changed the return type of 'RejectRequest*' methods to Exception and instead did `throw RejectRequest(...` ?